test/dotnet-add-proj.Tests/GivenDotnetAddProj.cs

```diff
@@ -0,0 +1,236 @@
+            cmd.StdErr.Should().BeEmpty();
+
+            var slnFile = new SlnFile();
+            slnFile.Read(Path.Combine(slnDir, slnName));
+            VerifyProjectInSolution(slnFile, projectName, projectGuid);
+
+            Restore(Path.Combine(setup.TestRoot, "App"), "App.csproj");
+            Build(slnDir, slnName);
+        }
+
+        [Fact]
+        public void WhenSolutionAlreadyContainsProjectItDoesntDuplicate()
+        {
+            var setup = CreateTestSetup(TestGroup, ProjectName);
+            string slnDir = Path.Combine(setup.TestRoot, "MoreThanOne");
+            string slnName = "AppAndLib.sln";
+            string slnFullPath = Path.Combine(slnDir, slnName);
+
+            var contentBefore = File.ReadAllText(slnFullPath);
+            var cmd = new AddProjCommand()
+                .WithWorkingDirectory(slnDir)
+                .WithSolution(slnName)
+                .Execute($"\"{setup.GetProjectFullPath("Lib")}\"");
+            cmd.Should().Pass();
+            cmd.StdOut.Should().Contain("already contains project");
+            cmd.StdErr.Should().BeEmpty();
+
+            var contentAfter = File.ReadAllText(slnFullPath);
+            contentAfter.Should().BeEquivalentTo(contentBefore);
+        }
+
+        [Fact]
+        public void WhenPassedMultipleProjectsAndOneOfthemDoesNotExistItCancelsWholeOperation()
+        {
+            var setup = CreateTestSetup(TestGroup, ProjectName);
+            string slnDir = Path.Combine(setup.TestRoot, "MoreThanOne");
+            string slnName = "App.sln";
+            string slnFullPath = Path.Combine(slnDir, slnName);
+
+            var contentBefore = File.ReadAllText(slnFullPath);
+            var cmd = new AddProjCommand()
+                .WithWorkingDirectory(slnDir)
+                .WithSolution(slnName)
+                .Execute($"\"{setup.GetProjectFullPath("Lib")}\" \"IDoNotExist.csproj\"");
+            cmd.Should().Fail();
+            cmd.StdErr.Should().Contain("does not exist");
+            cmd.StdErr.Should().NotMatchRegex("(.*does not exist.*){2,}");
+
+            var contentAfter = File.ReadAllText(slnFullPath);
+            contentAfter.Should().BeEquivalentTo(contentBefore);
+        }
+
+        [Fact]
+        public void WhenPassedProjectDoesNotExistAndForceSwitchIsPassedItAddsIt()
+        {
+            var setup = CreateTestSetup(TestGroup, ProjectName);
+            string slnDir = Path.Combine(setup.TestRoot, "MoreThanOne");
+            string slnName = "App.sln";
+            string slnFullPath = Path.Combine(slnDir, slnName);
+
+            var contentBefore = File.ReadAllText(slnFullPath);
+            var cmd = new AddProjCommand()
+                .WithWorkingDirectory(slnDir)
+                .WithSolution(slnName)
+                .Execute("--force \"IDoNotExist.csproj\"");
+            cmd.Should().Pass();
+            cmd.StdOut.Should().Contain("added to the solution");
+            cmd.StdErr.Should().BeEmpty();
```

Comment:
Help text said something about relative paths, but I don't see that reflected in the test...